### ISSUE TITLE
Password does not explain whats wrong

### ISSUE THREAD

**Issue by Zapnologica** [ragunathjawahar/android-saripaar#128]
I have the following in my Activity:
    
    @NotEmpty
    @Password(min = 6, scheme = Password.Scheme.ALPHA_NUMERIC)
    private EditText editTextPassword;

Now if I type a password that is 4 long.  My EditText just says Invalid Password. It does not tel the user exactly what is wrong?

here is my OnValidationFailed:

  @Override
    public void onValidationFailed(List<ValidationError> errors) {
        mTracker.send(new HitBuilders.EventBuilder()            
        for (ValidationError error : errors) {
            View view = error.getView();
            String message = error.getCollatedErrorMessage(this);

            // Display error messages ;)
            if (view instanceof EditText) {
                ((EditText) view).setError(message);
            } else {
                Toast.makeText(this, message, Toast.LENGTH_LONG).show();
            }
        }
    }

is there anything im missing.

**Comment by ragunathjawahar** [comment]
Yeah, currently the validator does not dynamically display error messages. You have to use a message like - Password should contain alphabets and numbers. That's the best we can do at the moment, however I'll keep this open as an enhancement.

**Comment by Zapnologica** [comment]
Good day,

i Just wanted to pop i and check what that status and possibility of adding the enhancement that explains what part of the validation failed from the annotations. For example.

if you set min length 3, there should be a standard error message saying the field needs to be more than 3 characters.

And then for passwords, based on the scheme which you chose in the annotation it should be able to show you what you are missing?

**Comment by ragunathjawahar** [comment]
@Zapnologica this enhancement would be addressed in the next release.

**Comment by forrestbice** [comment]
@ragunathjawahar related question:

Right now you can provide a custom message via:

`@Password(min = 8, scheme = Scheme.ALPHA_NUMERIC, message = "Your Custom Message"`

Is it possible to utilize a localized string? I want to access resources in my `strings.xml` via something like:

`@Password(min = 8, scheme = Scheme.ALPHA_NUMERIC, message = R.string.your_custom_message`

**Comment by forrestbice** [comment]
My apologies, I missed this comment (https://github.com/ragunathjawahar/android-saripaar/issues/117#issue-111121725) which indicates this is possible via messageResId!

**Comment by uchennafokoye** [comment]
Hi, I was wondering how this was addressed. I can't find any information on how to do multiple validation messages for different problems. Thanks!

**Comment by ragunathjawahar** [comment]
This feature isn't in the library yet. There is a workaround though, inside the `onValidationFailed(List<ValidationError>)` callback, check if the validation failed on the password field and evaluate the criteria that caused the failure and then display an appropriate message.
